src/java2yaml/Steps/GenerateToc.cs

```diff
@@ -32,14 +39,59 @@ public Task RunAsync(ConfigModel config)
                 else
                 {
                     MergeToc();
+
+                    ConsoleLogger.WriteLine(new LogEntry
+                    {
+                        Phase = StepName,
+                        Level = LogLevel.Info,
+                        Message = $" Merge completed, write toc.yml to {_config.OutputPath} ..."
+                    });
                 }
             }
             );
         }
 
         private void MergeToc()
         {
-            throw new NotImplementedException();
+            var tocFiles = new List<TocYaml>();
+
+            tocFiles = GetTocList()
+                .Select(file => DeserializeYaml(file))
+                .ToList();
+
+            WriteMergedTocToDisk(tocFiles);
+        }
+
+        private List<string> GetTocList()
+        {
+            return ( _config.RepositoryFolders
+                    .Select(path => FileUtility.GetFilesByName(Path.Combine(path, Constants.Doc), Constants.Toc).FirstOrDefault())
+                    .Select(x => x.FullName.ToString())

```

Comment:
`.Where(path => path != null)` ?

Reply:
expect the _config.RepositoryFolders is not null or empty, because:

1. _config.RepositoryFolders is a list of absolute path of source code folder we need to document, if it is null or empty, the program will be terminated in previous validation.
2. only read action to _config.RepositoryFolders, no write